client/src/state/toasts/ToastsActions.js

```diff
@@ -0,0 +1,106 @@
+import ACTION_TYPES from './ToastsActionTypes';
+import { FADEOUT_TIME } from './ToastConstants';
+import { v4 as uuid } from 'uuid';
+
+/**
+ * @typedef {import('./ToastsTypes').BasicToastOption} BasicToastOption
+ */
+
+/**
+ * @callback reduxThunk
+ * @param {function} dispatch
+ */
+
+/**
+ * @type BasicToastOption
+ */
+const defaultOptions = {
+  text: '',
+  stay: false,
+  type: 'notice'
+};
+
+/**
+ * Dismiss a toast notification based on its ID.
+ * @param {string} id
+ * @returns {reduxThunk}
+ */
+export function dismiss(id) {
+  return (dispatch) => {
+    dispatch({
+      type: ACTION_TYPES.DISMISS,
+      payload: { id }
+    });
+
+    setTimeout(
+      () => dispatch({ type: ACTION_TYPES.REMOVE, payload: { id } }),
+      FADEOUT_TIME
+    );
+  };
+}
+
+/**
+ * Display a toast notification
+ * @param {ToastOption} options
+ * @returns {reduxThunk}
+ */
+export function display(options) {
+  const id = `toast-${uuid()}`;
+
+  return (dispatch) => {
+    const dismissCallback = () => dismiss(id)(dispatch);
+    const payload = { id, dismissCallback, ...defaultOptions, ...options };
+
+    dispatch({ type: ACTION_TYPES.DISPLAY, payload });
+  };
+}
+
+/**
+ * Display a success toast
+ * @param {string} text
+ * @returns {reduxThunk}
+ */
+export function success(text) {
+  return display({ text, type: 'success' });
+}
+
+/**
+ * Display an error toast
+ * @param {string} text
+ * @returns {reduxThunk}
+ */
+export function error(text) {
+  return display({ text, type: 'error', stay: true });
```

Comment:
Is it worth making the `stay` argument available for devs to pass? I might not want my errors/warnings to stick around.

Reply:
`error`, `info`, `warning` and `success` are meant to be shorthand for `display`. That's why I made them as simple as possible, with the thinking that if you want a bit more control you should call display directly.

The rational for having warning/error stick around was that those are more important, so the message should stay there by default. notice/success can be missed or ignored without much downside, so it's OK to let them timeout.